src/WebJobs.Script/Workers/Rpc/FunctionRegistration/RpcFunctionInvocationDispatcher.cs

```diff
@@ -323,7 +323,7 @@ private async Task DisposeAndRestartWorkerChannel(string runtime, string workerI
                 {
                     _logger.LogDebug("Disposing JobHost channel for workerId: {channelId}, for runtime:{language}", workerId, runtime);
                     isJobHostChannel = true;
-                    _jobHostLanguageWorkerChannelManager.DisposeAndRemoveChannel(channel);
+                    _jobHostLanguageWorkerChannelManager.DisposeAndRemoveChannel(channel, workerException);
```

Comment:
TryFailExecutions is called in ShutdownChannelIfExistsAsync, correct?

Reply:
Called in `ShutdownChannelIfExistsAsync` for WebHostLanguageWorkerChannelManager and called in `DisposeAndRemoveChannel` in JobHostLanguageWorkerChannelManager